var/spack/repos/builtin/packages/py-ply/package.py

```diff
@@ -7,9 +7,10 @@
 
 
 class PyPly(PythonPackage):
-    """PLY is nothing more than a straightforward lex/yacc implementation."""
+    """Python Lex & Yacc."""
+
     homepage = "http://www.dabeaz.com/ply"
-    url      = "https://www.dabeaz.com/ply/ply-3.11.tar.gz"
+    pypi = "ply/ply-3.11.tar.gz"
 
     version('3.11', sha256='00c7c1aaa88358b9c765b6d3000c6eec0ba42abca5351b095321aef446081da3')
     version('3.8', sha256='e7d1bdff026beb159c9942f7a17e102c375638d9478a7ecd4cc0c76afd8de0b8')
```

Comment:
I think this should have a build dep on setuptools

Reply:
I did not add it because it falls back to using distutils if setuptools is not found and the setuptools dependency was not in there before. But sure, I can add it.

Reply:
That works for now, but distutils is deprecated in Python 3.10 and will be removed in 3.12, so everything is moving over to setuptools now.

Reply:
I did not know that. Thanks for the explanation.